docs/components/javascript.md

```diff
@@ -0,0 +1,78 @@
+# Component JavaScript
+
+
+## Architecture
+
+Each component is a function which accepts an object containing all of the properties needed to render it and returns information describing what to render. These are called "functional components" or "stateless functional components" (SFCs) by many frameworks.
+
+Following this pattern means that components authored with x-dash can be compatible with a variety of static and dynamic runtimes including React, Preact, Inferno, VDO, and Hyperons amongst others.
+
+Interactive components which change state in reaction to events can still be created using the [x-interaction] component.
+
+[x-interaction]: ../../components/x-interaction
+
+
+## Syntax
+
+You may use all syntax up to and including the ECMAScript 2018 specification and any features that may be polyfilled by the [Polyfill Service] in your component source code. We do not currently support transpiling any features at proposal, draft, or candidate stages (if you are unsure what this means check out the [TC39 Process document]).
```

Comment:
Could you elaborate on how to use the Polyfill Service? Can we assume it is available? Or should components indicate that they are relying on certain features? This discussion and more over here: https://github.com/Financial-Times/x-dash/pull/78#issuecomment-424346708